File: CodeChef/CodeChef Starters 39/Closest_Vowels.py
# CodeChef CLOSEVOWEL Closest Vowels
# https://www.codechef.com/problems/CLOSEVOWEL

import sys
input = sys.stdin.readline
MOD = 1000000007

T = int(input())

for _ in range(T):
    N = int(input())
    S = input().rstrip()

    ans = 1

    s = set("cglr")
    for c in S:
        if c in s:
            ans = (ans * 2) % MOD

    print(ans)

File: CodeChef/CodeChef Starters 39/Closest_Vowels.cpp
// CodeChef CLOSEVOWEL Closest Vowels
// https://www.codechef.com/problems/CLOSEVOWEL

#include "bits/stdc++.h"
using namespace std;
const int MOD = 1000000007;

void solve() {
    int N; cin >> N;
    string S; cin >> S;

    long long ans = 1;
    vector<char> special_chars = {'c', 'g', 'l', 'r'};

    for (char c: S) {
        if (find(special_chars.begin(), special_chars.end(), c) != special_chars.end()) {
            ans = (ans * 2) % MOD;
        }
    }

    cout << ans << endl;
}

int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);
    int T; cin >> T;
    while(T--) {
        solve();
    }
    

}
